### PR TITLE
Add some documentation on flash-flood (closes #78)

### LINKED ISSUE #78: Create eventing documentation

### PR DESCRIPTION
**Pull request by natanlao**

### CODE REVIEW

File: docs/flash-flood.ipynb
Comment (amarjandu) [inline docs/flash-flood.ipynb]: just a nit, could there be an end of line delimiter such as `\n`? makes it easier to see.

File: docs/flash-flood.md
Comment (amarjandu) [inline docs/flash-flood.md]: suggestion: `filtered using JMESpath`?

**Review by amarjandu — APPROVED** [review]
small nits, i would wait for @chmreid to review before merging.

File: docs/flash-flood.md
Comment (chmreid) [reply docs/flash-flood.md]: https://jmespath.org/ says JMESPath

**Review by chmreid — DISMISSED** [review]
I am going to contribute some specific suggestions using a PR based on this PR, since my suggested edits were too substantial for a code review.

The notebook is not specific to the data store - it is more general and reads a bit like an expanded flash-flood readme - so it is probably more appropriate for the flash-flood repository. When we discussed making a flash-flood consumer vignette notebook, I was envisioning something in the context of the data store (i.e., using flash-flood data from an HCA or platform-gi data store deploym,ent), rather than showing _general_ usage of flash-flood. (For example, if I wanted to build a search engine that would index data in the data store, what would the data look like? If I received an event from a flash-flood event stream telling me a bundle was uploaded, how would I access the metadata of the new bundle?)

Regarding the flash-flood markdown document, when explaining the "downstream component" idea that Brian articulated, I would leave Elasticsearch out of the discussion entirely - because it does not interact with flash flood in any way, and is (still) not a separate component, it is a red herring and will confuse the reader. Instead, I would mention a generic indexer or a generic "search engine" as the hypothetical downstream component.

Please reach out if you have questions about this.

**Comment by chmreid** [comment]
One other point I wanted to raise - this PR adds a jupyter notebook to the repo, but we don't have any other notebooks in this repo and we don't have jupyter notebook in any of our dependencies. 

I am reluctant to add a notebook  to this repo, for those reasons and because (as mentioned above) this notebook probably fits better for the flash-flood repo. If the notebook contained data store-specific demonstration of flash-flood usage with a data store deployment, then I'd definitely feel better about adding the first Jupyter notebook to the data store docs.

**Comment by natanlao** [comment]
> The notebook is not specific to the data store - it is more general and reads a bit like an expanded flash-flood readme - so it is probably more appropriate for the flash-flood repository.

My bad - I clearly misunderstood the intent of the original issue. I understand now that you wanted an application of flash flood to the data store and how it might be extended, not a summary of its functionality in the context of our last meeting.

> (For example, if I wanted to build a search engine that would index data in the data store, what would the data look like? If I received an event from a flash-flood event stream telling me a bundle was uploaded, how would I access the metadata of the new bundle?)

I'm happy to make the adjustments to `flash-flood.md` that you've outlined and wait for your PR as guidance.

> One other point I wanted to raise - this PR adds a jupyter notebook to the repo, but we don't have any other notebooks in this repo and we don't have jupyter notebook in any of our dependencies.

I always thought "vignette" was synonymous with "Jupyter notebook." Google shows that is not the case. I can convert this to just Markdown, or whichever you think is most appropriate.

**Pull request closed without merging**
